src/main/java/com/gmail/zlotnikova/repository/impl/ConnectionRepositoryImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.gmail.zlotnikova.repository.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.gmail.zlotnikova.repository.ConnectionRepository;
+import com.gmail.zlotnikova.util.PropertyUtil;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import static com.gmail.zlotnikova.repository.constant.ConnectionConstant.DATABASE_PASSWORD;
+import static com.gmail.zlotnikova.repository.constant.ConnectionConstant.DATABASE_URL;
+import static com.gmail.zlotnikova.repository.constant.ConnectionConstant.DATABASE_USERNAME;
+import static com.gmail.zlotnikova.repository.constant.ConnectionConstant.MYSQL_SERVER_URL;
+import static com.gmail.zlotnikova.repository.constant.DatasourcePropertyConstant.PREPARED_STATEMENTS_CACHE_ENABLE;
+import static com.gmail.zlotnikova.repository.constant.DatasourcePropertyConstant.PREPARED_STATEMENTS_CACHE_LIMIT;
+import static com.gmail.zlotnikova.repository.constant.DatasourcePropertyConstant.PREPARED_STATEMENTS_CACHE_SIZE;
+
+public class ConnectionRepositoryImpl implements ConnectionRepository {
+
+    private static ConnectionRepository instance;
+
+    private ConnectionRepositoryImpl() {
+    }
+
+    public static ConnectionRepository getInstance() {
+        if (instance == null) {
+            instance = new ConnectionRepositoryImpl();
+        }
+        return instance;
+    }
+
+    private static final Logger logger = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+    private static final String MYSQL_JDBC_DRIVER = "com.mysql.cj.jdbc.Driver";
+    private static HikariDataSource ds;
+
+    @Override
+    public Connection getInitialConnection() throws IllegalStateException {
```

Comment:
No need to use it if we use the connection pool, the database should be created before run app